app/tut/ga/scoip.md

```diff
@@ -0,0 +1,392 @@
+
+Mar shampla, féach ar an gcód seo:
+
+```{.setanta .numberLines}
+scríobh("Táim i mo chónaí i")
+tír := "Éire"
+scríobh(tír)
+```
+
+Is féidir linn an athróg `tír` a úsáid i líne a trí, mar tá líne a trí tar éis líne a dó. Níl an
+ríomhchlár seo a leanas ceart mar ní féidir linn an athróg a úsáid sula gcruthaímid é.
+
+```{.setanta .numberLines}
+scríobh("Táim i mo chónaí i")
+scríobh(tír)
+tír := "Éire"
+```
+
+[[Éiríonn rudaí níos casta|Things get more complex]] nuair a bhainimid úsáid as `má`{.setanta},
+`nuair-a`{.setanta}, `le idir`{.setanta} agus gníomhartha.
+
+Féach ar an gcód seo:
+
+```{.setanta .numberLines}
+má fíor {
+    x := 2
+}
+scríobh(x)
+```
+
+An bhfuil an cód sin ceart? Níl sé: Tá scóip `x` [[teoranta|limited]]. Nuair a chruthaíonn tú athróg
+i [[mbloc cóid|block of code]] (cód idir `{` agus `}`), ní féidir leat an athróg sin a úsáid
+[[ón taobh amuigh|from outside]]. Áfach, is féidir leat é a úsáid i mbloc cóid eile atá
+[[neadaithe|nested]] isteach sa bhloc, mar seo:
+
+```{.setanta .numberLines}
+má fíor {
+    x := 2
+    le i idir (0, 2) {
+        scríobh(x + i)
+    }
+}
+```
+
+Tá na rialacha seo [[i gceist le|apply to]] gníomhartha agus creatlacha freisin. Mar shampla,
+smaoinigh ar an ríomhchlár seo: Cad a scríobhfaidh sé?
+
+```{.setanta .numberLines}
+ainm := "Niamh"
+gníomh dia_duit() {
+    scríobh("Dia duit", ainm)
+}
+
+dia_duit()
+
+ainm = "Oisín"
+
+dia_duit()
+
+má 1 == 1 {
+   ainm := "Fionn" 
+
+   dia_duit()
+}
+```
+
+Bain triail as!
+
+{{{
+ainm := "Niamh"
+gníomh dia_duit() {
+    scríobh("Dia duit", ainm)
+}
+
+dia_duit()
+
+ainm = "Oisín"
+
+dia_duit()
+
+má 1 == 1 {
+   ainm := "Fionn" 
+
+   dia_duit()
+}
+}}}
+
+An raibh tú ceart? Oibríonn an líne `ainm = "Oisín"`{.setanta} ach ní dhéanann an líne `ainm :=
+"Fionn"`{.setanta} aon rud. Tharlaíonn é sin mar tá `ainm` sa ghníomh `dia_duit` i scóip an chéad
+athróg `ainm`, agus cruthaíonn an líne `ainm := "Fionn"`{.setanta} athróg eile le scóip eile: Níl
+aon [[tionchar|effect]] aige.
+
+Sa pictiúr seo, tá dath difriúil bainte le gach athróg ar leith, tabhair faoi deara ní úsáidtear an
+dara athróg `ainm` in aon áit.
+
+![Scóip Ghnímh](assets/scoip-ghnimh.png)
+
+# Scáthú
+
+Cad a tharlaíonn nuair a cruthaímid dhá athróg leis an ainm céanna? Chonaiceamar é seo sa ríomhchlár
+ag barr an leathanach:
+
+```{.setanta .numberLines}
+x := 2
+má x > 1 {
+    x := 3
+    scríobh(x)
+}
+scríobh(x)
+```
+
+Tá an 4ú líne (`scríobh(x)`{.setanta}) i scóip an dhá athróg `x`; An ceann ar an gcéad líne agus an
+ceann ar an 3ú líne. Sa chás seo úsáideann sé an ceann [[is cóngaraí|closest]] (an ceann sa bhloc
+[[is faide isteach|innermost]]).
+
+Tá sé cosúil go cuireann an dara athróg an chéad cheann [[i bhfolach|hides]]. Tugaimid
+[[**scathú**|shadowing]] ar an tionchar seo mar nuair a cruthaímid an athróg nua agus úsáidimid ainm
+atá ag athróg eile cheana, ní féidir linn an sean athróg a fheiceáil [[a thuilleadh|anymore]].
+
```

Comment:
B'fhiú a rá, is dóigh liom, gur drochchleachtas ar cheart é a sheachaint athróg a chruthú a fholaíonn athróg eile - gur ceart ainmneacha a roghnú go  cúramach.

Reply:
Chuir mé téacs mar sin isteach, GRMA